File: backend/app/routers/media_sales.py
from fastapi import APIRouter, HTTPException
from app.models.media_sales import NewMediaSalesSnapshot, MediaSalesSnapshot
from app.services import media_sales_sheets

router = APIRouter(prefix="/media-sales", tags=["media-sales"])


@router.get("/snapshots", response_model=list[MediaSalesSnapshot])
def list_snapshots():
    try:
        return media_sales_sheets.get_all_snapshots()
    except Exception as exc:
        raise HTTPException(status_code=500, detail=f"Could not fetch media-sales snapshots: {exc}")


@router.post("/snapshots", status_code=201)
def add_snapshot(payload: NewMediaSalesSnapshot):
    try:
        media_sales_sheets.append_snapshot(
            date=payload.date,
            channel_sponsors=payload.channel_sponsors,
            ad_revenue_per_sponsor=payload.ad_revenue_per_sponsor,
        )
        return {"message": "Snapshot added"}
    except Exception as exc:
        raise HTTPException(status_code=500, detail=f"Could not add snapshot: {exc}")


@router.delete("/snapshots/{row_index}", status_code=204)
def remove_snapshot(row_index: int):
    try:
        media_sales_sheets.delete_snapshot(row_index)
    except Exception as exc:
        raise HTTPException(status_code=500, detail=f"Could not delete snapshot: {exc}")

File: frontend/src/api/mediaSales.ts
import { api } from "./client";

export interface MediaSalesSnapshot {
  row_index: number;
  date: string;
  channel_sponsors: number;
  ad_revenue_per_sponsor: number;
}

export interface NewMediaSalesSnapshot {
  date: string;
  channel_sponsors: number;
  ad_revenue_per_sponsor: number;
}

export async function getMediaSalesSnapshots(): Promise<MediaSalesSnapshot[]> {
  const { data } = await api.get<MediaSalesSnapshot[]>("/media-sales/snapshots");
  return data;
}

export async function addMediaSalesSnapshot(snapshot: NewMediaSalesSnapshot): Promise<void> {
  await api.post("/media-sales/snapshots", snapshot);
}

export async function deleteMediaSalesSnapshot(rowIndex: number): Promise<void> {
  await api.delete(`/media-sales/snapshots/${rowIndex}`);
}
